packages/components/src/Combobox/components/ComboboxContent/ComboboxHeader.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { Typography } from "@jobber/components/Typography";
+import { Button } from "@jobber/components/Button";
+import styles from "./ComboboxHeader.css";
+
+interface ComboboxHeaderProps {
+  readonly subjectNoun?: string;
+  readonly selectedCount: number;
+  readonly onClearAll: () => void;
+  readonly onSelectAll: () => void;
+}
+
+export function ComboboxHeader(props: ComboboxHeaderProps): JSX.Element {
+  const hasSelected = props.selectedCount > 0;
+  let label = "Select";
+  let actionLabel = "Select all";
```

Comment:
we've got a few bits of logic here that can be thematically grouped around labels

thinking ahead, as a courtesy to future us (or other developers) let's try to extract this out to a function so that we have as much room as possible before hitting the ol' `Function ComboboxHeader has too many statements (13)` at which point someone will be forced to refactor things

what do you think about this?

```
  const hasSelected = props.selectedCount > 0;
  const actionLabel = hasSelected ? "Clear" : "Select all";
  const label = getLabel(hasSelected, props.selectedCount, props.subjectNoun);
  const handleSelectAll = hasSelected ? props.onClearAll : props.onSelectAll;
```

where `getLabel` is a function defined outside the body of the `ComboboxHeader` to avoid re-making it

```
function getLabel(
  hasSelected: boolean,
  count: number,
  subjectNoun?: string,
): string {
  if (hasSelected) {
    return `${count} selected`;
  }

  if (subjectNoun) {
    return `Select ${subjectNoun}`;
  }

  return "Select";
}
```

Reply:
admittedly this is a bit heavy on ternaries, but I am of the opinion these are pretty easy to understand - so they're an appropriate use that keeps the code lean and doesn't sacrifice readability

Reply:
Great call out, thank you Zak! I fixed this here 82d4f425d907227f5b358cd59976fb8afc98563c
I was previously heavy on ternaries in this file so went the opposite way.  Because these ternaries are 1 line I'm totally fine with that for the sake of keeping things lean